File: openHAB/OpenHABLinkedPage.m
//
//  OpenHABLinkedPage.m
//  HelloRestKit
//
//  Created by Victor Belov on 10/01/14.
//  Copyright (c) 2014 Victor Belov. All rights reserved.
//

#import "OpenHABLinkedPage.h"
#import "GDataXMLNode.h"

@implementation OpenHABLinkedPage
@synthesize pageId, title, icon, link;

- (OpenHABLinkedPage *) initWithXML:(GDataXMLElement *)xmlElement
{
    self = [super init];
    for (GDataXMLElement *child in [xmlElement children]) {
        if (![[child name] isEqual:@"id"]) {
            if ([[self allPropertyNames] containsObject:[child name]])
                [self setValue:[child stringValue] forKey:[child name]];
        } else {
            pageId = [child stringValue];
        }
    }
    return self;
}

- (OpenHABLinkedPage *) initWithDictionary:(NSDictionary *)dictionary
{
    self = [super init];
    for (NSString *key in [dictionary allKeys]) {
        if (![key isEqualToString:@"id"]) {
            if ([[self allPropertyNames] containsObject:key]) {
                [self setValue:[dictionary objectForKey:key] forKey:key];
            }
        } else {
            pageId = [dictionary objectForKey:key];
        }
    }
    return self;
}

@end

File: openHAB/OpenHABLinkedPage.swift
// Copyright (c) 2010-2019 Contributors to the openHAB project
//
// See the NOTICE file(s) distributed with this work for additional
// information.
//
// This program and the accompanying materials are made available under the
// terms of the Eclipse Public License 2.0 which is available at
// http://www.eclipse.org/legal/epl-2.0
//
// SPDX-License-Identifier: EPL-2.0

import Foundation
import Fuzi

class OpenHABLinkedPage: NSObject, Decodable {
    private enum CodingKeys: String, CodingKey {
        case pageId = "id"
        case title
        case icon
        case link
    }

    var pageId = ""
    var title = ""
    var icon = ""
    var link = ""

    init(xml xmlElement: XMLElement) {
        super.init()
        for child in xmlElement.children {
            switch child.tag {
            case "title": title = child.stringValue
            case "icon": icon = child.stringValue
            case "link": link = child.stringValue
            case "id": pageId = child.stringValue
            default:
                break
            }
        }
    }
}
